dvc/repo/scm_context.py

```diff
@@ -1,13 +0,0 @@
-def scm_context(method):
-    def run(repo, *args, **kw):
-        try:
-            result = method(repo, *args, **kw)
-            repo.scm.reset_ignores()
-            repo.scm.remind_to_track()
-            repo.scm.reset_tracked_files()
-            return result
-        except Exception:
-            repo.scm.cleanup_ignores()
-            raise
-
-    return run
```

Comment:
Why do you merge this with new context? Looks like this one is not always needed.